### ISSUE TITLE
Possible false positives re: deftheme

### ISSUE THREAD

**Issue by riscy** [purcell/package-lint#208]
Hello!  I believe in certain cases false positives are being raised with the use of deftheme regarding the package prefix.

Here's a minimal file (that would be named riscy-theme.el):
```elisp
;;; riscy-theme.el --- Riscy theme

;; Author: riscy
;; Version: 1.0
;; URL: https://github.com/riscy
;; Package-Requires: ((emacs "26.1"))

;;; Commentary:

;; ...

;;; Code:

(deftheme riscy "...")

(custom-theme-set-faces 'riscy nil)

(provide-theme 'riscy)
(provide 'riscy-theme)
;;; riscy-theme.el ends here
```

This file raises the following package-lint (using version 20210320.438):
```
14:0: error: "riscy" doesn't start with package's prefix "riscy-theme".
```
but I believe, per convention, this is exactly what the theme should be called in this case.  This cropped up in a recent MELPA PR here: https://github.com/melpa/melpa/pull/7520#issuecomment-822160187

**Comment by rgrmrts** [comment]
I'm encountering the same thing for the same scenario, also I believe themes do not need the following:
```elisp
(provide '<name>-theme)
```
and having `(provide-theme '<name>)` is convention. Running package lint on a theme I'm working on suggests adding that as well.

**Comment by purcell** [comment]
I believe I have done this now, thanks @riscy. Let me know if you find any issues.

**Comment by rgrmrts** [comment]
Confirming that your latest changes fixed the error described in this issue. Thanks @purcell!

**Comment by purcell** [comment]
Thanks @rgrmrts 🤜🤛
